[fx] import used modules for code gen

zdevito opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #45686 [packaging] simpler dependency plotting
* **#45471 [fx] import used modules for code gen**

Intead of assuming that 'torch' is the only module used by generated code,
use the qualified names of builtin functions to generate import statements
for all builtins. This allows user-captured functions to also get code generated correctly.

Differential Revision: [D23978696](https://our.internmc.facebook.com/intern/diff/D23978696)

zdevito committed:
[fx] import used modules for code gen

Intead of assuming that 'torch' is the only module used by generated code,
use the qualified names of builtin functions to generate import statements
for all builtins. This allows user-captured functions to also get code generated correctly.

[ghstack-poisoned]

jamesr66a reviewed (approved):
Looks fine. Add a test?

zdevito committed:
Update on "[fx] import used modules for code gen"


Intead of assuming that 'torch' is the only module used by generated code,
use the qualified names of builtin functions to generate import statements
for all builtins. This allows user-captured functions to also get code generated correctly.

Differential Revision: [D23978696](https://our.internmc.facebook.com/intern/diff/D23978696)

[ghstack-poisoned]

zdevito committed:
Update on "[fx] import used modules for code gen"


Intead of assuming that 'torch' is the only module used by generated code,
use the qualified names of builtin functions to generate import statements
for all builtins. This allows user-captured functions to also get code generated correctly.

Differential Revision: [D23978696](https://our.internmc.facebook.com/intern/diff/D23978696)

[ghstack-poisoned]

zdevito committed:
Update on "[fx] import used modules for code gen"


Intead of assuming that 'torch' is the only module used by generated code,
use the qualified names of builtin functions to generate import statements
for all builtins. This allows user-captured functions to also get code generated correctly.

Differential Revision: [D23978696](https://our.internmc.facebook.com/intern/diff/D23978696)

[ghstack-poisoned]

zdevito committed:
Update on "[fx] import used modules for code gen"


Intead of assuming that 'torch' is the only module used by generated code,
use the qualified names of builtin functions to generate import statements
for all builtins. This allows user-captured functions to also get code generated correctly.

Differential Revision: [D23978696](https://our.internmc.facebook.com/intern/diff/D23978696)

[ghstack-poisoned]

zdevito committed:
Update on "[fx] import used modules for code gen"


Intead of assuming that 'torch' is the only module used by generated code,
use the qualified names of builtin functions to generate import statements
for all builtins. This allows user-captured functions to also get code generated correctly.

Differential Revision: [D23978696](https://our.internmc.facebook.com/intern/diff/D23978696)

[ghstack-poisoned]

zdevito committed:
Update on "[fx] import used modules for code gen"


Intead of assuming that 'torch' is the only module used by generated code,
use the qualified names of builtin functions to generate import statements
for all builtins. This allows user-captured functions to also get code generated correctly.

Differential Revision: [D23978696](https://our.internmc.facebook.com/intern/diff/D23978696)

[ghstack-poisoned]
